server/src/main/java/io/druid/segment/realtime/appenderator/Appenderator.java

```diff
@@ -53,6 +53,14 @@
   Object startJob();
 
   /**
+   * Returns the size of data currently stored in local storage.  The persistedBytes is changed when
+   * {@link #persistAll(Committer)} and {@link #drop(SegmentIdentifier)} is called.
```

Comment:
Not strictly true, since persistAll will return before persistedBytes is updated (the persisting happens in another thread). That might lead to persisting more data than we actually want to persist, since we're not counting either the following:

- data in memory in the currently active IncrementalIndex
- data in IncrementalIndexes that are pending persist in the persist thread

I'm not sure what's best to do here. Maybe we can track both rows and bytes and have a trigger to publish if either one exceeds a limit?